src/util/sandbox/test_sandbox.c

```diff
@@ -0,0 +1,396 @@
+#ifndef FD_HAS_SANDBOX
+#define FD_HAS_SANDBOX
+#endif
+
+#ifdef FD_HAS_SANDBOX
+
+#define _GNU_SOURCE
+
+#include "../fd_util.h"
+#include "fd_sandbox.h"
+#include "fd_sandbox_private.h"
```

Comment:
Aspirationally, only the top level include (e.g. fd_util.h) would be necessary for a unit tester and and the APIs to test would be such that all functionality is testable public facing APIs.

Given all the low level OS hacker, this is probably not possible here.  But, we probably can get very close and at a minimum here, we shouldn't have to include fd_sandbox.h and fd_sandbox_private.h  (e.g. would it ever make sense to just include fd_sandbox_private.h standalone ... doubt it ... generally private APIs imply the public API will be available too).